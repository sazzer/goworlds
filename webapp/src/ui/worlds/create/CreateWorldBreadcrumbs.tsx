import React, {FunctionComponent} from 'react';
import {Breadcrumb} from "semantic-ui-react";
import {Link} from "react-router-dom";
import {useTranslation} from "react-i18next";

/** The type for the breadcrumbs */
export declare type CreateWorldBreadcrumbsProps = {
};

/**
 * Breadcrumbs to render on the Create World page
 * @constructor
 */
export const CreateWorldBreadcrumbs: FunctionComponent<CreateWorldBreadcrumbsProps> = () => {
    const { t } = useTranslation();

    return (
        <Breadcrumb>
            <Breadcrumb.Section>
                <Link to="/">{t('worlds.create.breadcrumbs.home')}</Link>
            </Breadcrumb.Section>
            <Breadcrumb.Divider />
            <Breadcrumb.Section active>{t('worlds.create.breadcrumbs.label')}</Breadcrumb.Section>
        </Breadcrumb>
    );
}
