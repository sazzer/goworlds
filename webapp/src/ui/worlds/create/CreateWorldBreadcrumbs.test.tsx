import React from 'react';
import {CreateWorldBreadcrumbs} from './CreateWorldBreadcrumbs';
import {render} from '@testing-library/react';
import {MemoryRouter} from "react-router";

/** Set up the component to test */
function setup() {
    return {
        element: render(<MemoryRouter><CreateWorldBreadcrumbs /></MemoryRouter>)
    };
}

it('renders without crashing', () => {
    const {element} = setup();

    expect(element.baseElement).toMatchSnapshot();
});
