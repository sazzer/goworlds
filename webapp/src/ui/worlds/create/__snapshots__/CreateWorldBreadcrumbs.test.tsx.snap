// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<body>
  <div>
    <div
      class="ui breadcrumb"
    >
      <div
        class="section"
      >
        <a
          href="/"
        >
          Home
        </a>
      </div>
      <div
        class="divider"
      >
        /
      </div>
      <div
        class="active section"
      >
        !!worlds.create.breadcrumbs.label!!
      </div>
    </div>
  </div>
</body>
`;
